omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java

```diff
@@ -256,59 +262,81 @@ public static void setFullySpecifiedName(Concept instance, String name) {
 		ConceptName fullySpecifiedName = new ConceptName(name, Context.getLocale());
 		instance.setFullySpecifiedName(fullySpecifiedName);
 	}
-	
 	/**
-	 * It's needed, because of ConversionException: Don't know how to handle collection class:
-	 * interface java.util.Collection
-	 * 
-	 * if request to update Concept updates ConceptName, adequate resource takes care of it,
-	 * so this method just adds new and removes deleted names.
 	 * 
 	 * @param instance
-	 * @param names
+	 * @param propertyName name of property to set
+	 * @param items objects to be added to property
+	 * @param itemsComparator is used to compare new and old list element's, which determines if item is added/removed
+	 * @return 
+	 * @throws IllegalAccessException
+	 * @throws InvocationTargetException
+	 * @throws NoSuchMethodException
 	 */
-	@PropertySetter("names")
-	public static void setNames(Concept instance, List<ConceptName> names) {
-		for(ConceptName name : names){
-			name.setConcept(instance);
+	protected static <T>void setHibernateCollectionProperty(Concept instance, String propertyName, Collection<T> items, Comparator<T> itemsComparator) 

```

Comment:
I agree we don't gain much here other than a nice coding exercise :) It's unlikely for those methods to be changed so I think it's ok to leave it this way.